lib/streamlit/watcher/file_watcher.py

```diff
@@ -31,6 +31,11 @@
     watchdog_available = True
 except ImportError:
     watchdog_available = False
+    # Stub the EventBasedFileWatcher so it can be mocked by tests
+
+    class EventBasedFileWatcher:  # type: ignore
+        pass
+
 
 # EventBasedFileWatcher won't be available if its import failed (due to
```

Comment:
nit: would be good to reword this slightly to mention that the real `EventBasedFileWatcher` won't be available without watchdog since we now stub it above

Reply:
Updated

Reply:
Oh one last thing:

> EventBasedFileWatcher won't be a stub only if its import failed...

Shouldn't this be "EventBasedFileWatcher will be a stub if"?

Reply:
Finally got back around to updating this.

Reply:
Thanks!